homeassistant/components/v2c/sensor.py

```diff
@@ -0,0 +1,104 @@
+"""Support for V2C EVSE sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from pytrydan import TrydanData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import V2CUpdateCoordinator
+from .entity import V2CBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class V2CPowerRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[TrydanData], float]
```

Comment:
This makes me wonder, why return a dict in the coordinators, while you're still going to use the lib directly anyway